### PR TITLE
CIV-4665: DQ - FT - Have you already sent expert reports

### PR DESCRIPTION
**Pull request by kenneth-hmcts**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-4665

### Change description ###
This PR contains the change by adding a page asking expert report is sent or not

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/views/features/directionsQuestionnaire/expert-reports.njk [hunk 1]
```diff
@@ -0,0 +1,66 @@
+{% extends "claim-details-tpl.njk" %}
+{% from "govuk/components/button/macro.njk" import govukButton %}
+{% from "govuk/components/inset-text/macro.njk" import govukInsetText %}
```
Comment (commitBlob) [inline src/main/views/features/directionsQuestionnaire/expert-reports.njk:3]:
```
{% from "govuk/components/inset-text/macro.njk" import govukInsetText %}
{% from "govuk/components/textarea/macro.njk" import govukTextarea %}
```

you don't need these imports

File: src/main/routes/features/directionsQuestionnaire/expertReportsController.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  saveExpertReports,
+} from '../../../services/features/directionsQuestionnaire/expertReportsService';
+import {constructResponseUrlWithIdParams} from '../../../common/utils/urlFormatter';
+
+const expertReportsController = express.Router();
+const expertReportsViewPath = 'features/directionsQuestionnaire/expert-reports';
+
+function renderView(form: GenericForm<ExpertReports>, res: express.Response): void {
+
+  const expertReportsForm = Object.assign(form);
```
Comment (commitBlob) [inline src/main/routes/features/directionsQuestionnaire/expertReportsController.ts:17]:
```
 const expertReportsForm = Object.assign(form);
 expertReportsForm.option = form.model.option;
```
I think you don't need to do this for your case, would you mind trying to remove it please and check whether selection is persisted after save.
Expected behaviour:
Select an option, save and continue, come back to the page and selection should be persisted.

If above doesn't happen then keep these two lines.

File: src/main/views/features/directionsQuestionnaire/expert-reports.njk (hunk 1 above)
Comment (kenneth-hmcts) [reply src/main/views/features/directionsQuestionnaire/expert-reports.njk]: Unused import have been removed

File: src/main/routes/features/directionsQuestionnaire/expertReportsController.ts (hunk 2 above)
Comment (kenneth-hmcts) [reply src/main/routes/features/directionsQuestionnaire/expertReportsController.ts]: Codes have been changed for that part. Checked with same as the expected behaviour.

**Review by commitBlob — APPROVED** [review]

File: src/main/services/features/directionsQuestionnaire/expertReportsService.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  try {
+    const caseData = await getCaseDataFromStore(claimId);
+    return caseData?.directionQuestionnaire?.expertReports ? caseData.directionQuestionnaire.expertReports : new ExpertReports();
+  } catch (error) {
+    logger.error(error);
+    throw error;
+  }
+};
+
+const getExpertReportsForm = (expertReports: ExpertReportsOptions): ExpertReports => {
```
Comment (GorilaNaranja) [inline src/main/services/features/directionsQuestionnaire/expertReportsService.ts:18]: I think this function should return GenericForm type so we won't need this line in the controller 
`const form = new GenericForm(expertReports);`

File: src/main/services/features/directionsQuestionnaire/expertReportsService.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  return new ExpertReports(expertReports);
+};
+
+const saveExpertReports = async (claimId: string, expertReports: ExpertReports) => {
+  try {
+    const caseData = await getCaseDataFromStore(claimId);
+    caseData.directionQuestionnaire = {...caseData.directionQuestionnaire};
```
Comment (GorilaNaranja) [inline src/main/services/features/directionsQuestionnaire/expertReportsService.ts:25]: You can try to use this instead lines 25 and 26
`  caseData.directionQuestionnaire = {...caseData.directionQuestionnaire, expertReports};
`

**Review by GorilaNaranja — APPROVED** [review]

File: src/main/services/features/directionsQuestionnaire/expertReportsService.ts (hunk 3 above)
Comment (kenneth-hmcts) [reply src/main/services/features/directionsQuestionnaire/expertReportsService.ts:18]: After discussion, there will be more changes if we change to generic form. So keep the same.

File: src/main/services/features/directionsQuestionnaire/expertReportsService.ts (hunk 4 above)
Comment (kenneth-hmcts) [reply src/main/services/features/directionsQuestionnaire/expertReportsService.ts]: Changes have been made. Retest has been done and same behaviour as expected

**Review by commitBlob — APPROVED** [review]

**Review by GorilaNaranja — APPROVED** [review]

File: src/main/services/features/directionsQuestionnaire/sentExpertReportsService.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import {SentExpertReports} from '../../../common/models/directionsQuestionnaire/sentExpertReports';
```
Comment (laarmada) [inline src/main/services/features/directionsQuestionnaire/sentExpertReportsService.ts:1]: you can remove that class and the related test

File: src/main/routes/features/directionsQuestionnaire/sentExpertReportsController.ts [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  try {
+    const claimId = req.params.id;
+    const expertReports = getSentExpertReportsForm(req.body.sentExpertReportsOptions);
+    const form = new GenericForm(expertReports);
+    form.validateSync();
+
+    if (form.hasErrors()) {
+      renderView(form, res);
+    } else {
+      await saveSentExpertReports(claimId, expertReports);
```
Comment (laarmada) [inline src/main/routes/features/directionsQuestionnaire/sentExpertReportsController.ts:37]: use save generic directionQuestionaire

File: src/main/routes/features/directionsQuestionnaire/sentExpertReportsController.ts [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    renderView(new GenericForm(await getSentExpertReports(req.params.id)), res);
+  } catch (error) {
+    next(error);
+  }
+});
+
+sentExpertReportsController.post(DQ_SENT_EXPERT_REPORTS_URL, async (req, res, next) => {
+  try {
+    const claimId = req.params.id;
+    const expertReports = getSentExpertReportsForm(req.body.sentExpertReportsOptions);
```
Comment (laarmada) [inline src/main/routes/features/directionsQuestionnaire/sentExpertReportsController.ts:30]: use getform generic directionQuestionaire

File: src/main/routes/features/directionsQuestionnaire/sentExpertReportsController.ts [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+const sentExpertReportsController = express.Router();
+const expertReportsViewPath = 'features/directionsQuestionnaire/sent-expert-reports';
+
+function renderView(form: GenericForm<SentExpertReports>, res: express.Response): void {
+  res.render(expertReportsViewPath, {form});
+}
+
+sentExpertReportsController.get(DQ_SENT_EXPERT_REPORTS_URL, async (req, res, next) => {
+  try {
+    renderView(new GenericForm(await getSentExpertReports(req.params.id)), res);
```
Comment (laarmada) [inline src/main/routes/features/directionsQuestionnaire/sentExpertReportsController.ts:21]: use get generic directionQuestionaire

**Review by laarmada — CHANGES_REQUESTED** [review]

**Review by laarmada — APPROVED** [review]

**Pull request merged**